### PR TITLE
Electrum import export

### PR DESCRIPTION
**Pull request by ben-kaufman**
With this PR you can export wallets to electrum. Even though this was possible before resulting in watch-only-wallets, you now get regular wallets as well if your device ist supported in electrum.

### CODE REVIEW

File: src/cryptoadvance/specter/helpers.py [hunk 1]
```diff
@@ -225,3 +225,57 @@ def generate_mnemonic(strength=256):
     mnemo = Mnemonic("english")
     words = mnemo.generate(strength=strength)
     return words
+
+
+def parse_wallet_data_import(wallet_data):
+    cosigners_types = []
```
Comment (k9ert) [inline src/cryptoadvance/specter/helpers.py:231]: Can you add a description here? What is that function doing?

File: src/cryptoadvance/specter/wallet.py [hunk 2]
```diff
@@ -791,47 +791,69 @@ def get_address_info(self, address):
         except:
             return None
 
-    def get_electrum_watchonly(self):
+    def get_electrum_file(self):
+        electrum_devices = [
```
Comment (k9ert) [inline src/cryptoadvance/specter/wallet.py:795]: Also some documentation here would be good as well. What is that function doing? What is it returning? Maybe you can add links to specifications of a "electrum-file"?

**Comment by k9ert** [comment]
I'm wondering how we could potentially tear down the lengthy controller.py. I just had a look at the urls and after sorting, this is the result:
```
➜  specter git:(refactor-startup) ✗ cat controller.py | grep app.route | sort
@app.route("/")
@app.route("/about")
@app.route("/devices/<device_alias>/", methods=["GET", "POST"])
@app.route("/device_setup_wizard/", methods=["GET", "POST"])
@app.route("/generatemnemonic/", methods=["GET", "POST"])
@app.route("/get_fee/<blocks>")
@app.route("/get_scantxoutset_status")
@app.route("/get_txout_set_info")
@app.route("/login", methods=["GET", "POST"])
@app.route("/logout", methods=["GET", "POST"])
@app.route("/new_device/", methods=["GET", "POST"])
@app.route("/new_wallet/")
@app.route("/new_wallet/<wallet_type>/", methods=["GET", "POST"])
@app.route("/register", methods=["GET", "POST"])
@app.route("/setprice/", methods=["GET", "POST"])
@app.route("/settings/auth", methods=["GET", "POST"])
@app.route("/settings/bitcoin_core", methods=["GET", "POST"])
@app.route("/settings/general", methods=["GET", "POST"])
@app.route("/settings/hwi", methods=["GET", "POST"])
@app.route("/settings/", methods=["GET"])
@app.route("/specter_backup.zip")
@app.route("/toggleshowprice/", methods=["GET", "POST"])
@app.route("/wallets_overview/")
@app.route("/wallets/<wallet_alias>/")
@app.route("/wallets/<wallet_alias>/broadcast/", methods=["GET", "POST"])
@app.route("/wallets/<wallet_alias>/combine/", methods=["POST"])
@app.route("/wallets/<wallet_alias>/decoderawtx/", methods=["GET", "POST"])
@app.route("/wallets/<wallet_alias>/receive/", methods=["GET", "POST"])
@app.route("/wallets/<wallet_alias>/get_wallet_rescan_progress", methods=["GET", "POST"]
@app.route("/wallets/<wallet_alias>/send")
@app.route("/wallets/<wallet_alias>/send/import")
@app.route("/wallets/<wallet_alias>/send/new", methods=["GET", "POST"])
@app.route("/wallets/<wallet_alias>/send/pending/", methods=["GET", "POST"])
@app.route("/wallets/<wallet_alias>/settings/", methods=["GET", "POST"])
@app.route("/wallets/<wallet_alias>/tx/")
@app.route("/wallets/<wallet_alias>/tx/history/")
@app.route("/wallets/<wallet_alias>/tx/utxo/", methods=["GET", "POST"])
```
I don't think we have to do that within PRs like this but i also think that this would be a good idea. 
If i would put that in a catchy prase, maybe:
"Do the refactorings within the same feature-implementation-work which make these refactorings necessary".
This clearly works better when the necessary refactorings are smaller, though.

File: src/cryptoadvance/specter/wallet.py (hunk 2 above)
Comment (ben-kaufman) [reply src/cryptoadvance/specter/wallet.py:795]: I can't find such specification for the file format :(

**Comment by ben-kaufman** [comment]
I'd be happy to break the controller file down, been on my to-do list for months... but I'll work on that in another dedicated PR

**Comment by ben-kaufman** [comment]
Following this up with #726 , let me know what you think @k9ert

**Comment by k9ert** [comment]
Have a look into my commit. If you put the comments like that, you can hover in your IDE over the function and it reveasl the documentation for that function.
I can't yet test that right now, i'll try to test tomorrow or thursday.
BTW, how did you figure out the electrum json-format? By exporting and then reimporting?

**Comment by ben-kaufman** [comment]
Yes, I just tried exporting and importing test wallets.

**Comment by k9ert** [comment]
Not sure how to test this. I clicked on "Download Electrum Watch-only-File" and tried to load that file via electrum->File->Open.
As a result i get:
image

**Comment by k9ert** [comment]
The error-message was due to a mismatch of networks: Exported a regtest wallet but imported in a mainnet electrum.
Starting electrum with `--testnet` switch made the test work.

**Pull request merged**